### PR TITLE
Transfers and manage

### PR DESCRIPTION
**Pull request by JTalarico**
Deposit is not finished at all. Withdraw is mostly done. Havent figured out email validation (the regex im using never works.) Send money is done. I'm just waiting on the contacts to be finished.

### CODE REVIEW

File: src/pages/manage/deposit/deposit.component.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { BankTransferService } from "../../../providers/bank.service";
+import { Storage } from "@ionic/storage";
+import { WalletsService } from "../../../providers/wallet.service";
+import {Validators, FormBuilder, FormGroup} from "@angular/forms";
```
Comment (ghiobi) [inline src/pages/manage/deposit/deposit.component.ts]: Please put a space here `{ Validator...`

File: src/pages/manage/deposit/deposit.component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      .then(currencies => {
+        this.currencies = currencies;
+        this.form.value.currency = this.currencies[0];
+        this.setDecimalPlaces();
+      });
+
+    this.wallets = this.walletSrv.wallets
+  }
+
+  public setDecimalPlaces() {
```
Comment (ghiobi) [inline src/pages/manage/deposit/deposit.component.ts]: TsDoc is missing

File: src/pages/manage/deposit/deposit.component.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    private formBuilder: FormBuilder
+
+  ) {
+    this.form = this.formBuilder.group({
+      amount: ['', Validators.required],
+      currency: ['', Validators.required]
+    });
+  }
+
+  ionViewDidLoad() {
```
Comment (ghiobi) [inline src/pages/manage/deposit/deposit.component.ts]: TsDoc is missing

File: src/pages/manage/withdraw/withdraw.component.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit } from '@angular/core';
+import { AlertController, LoadingController, Loading } from "ionic-angular";
+import { BankTransferService } from "../../../providers/bank.service";
+import { WalletsService } from "../../../providers/wallet.service";
+import {Alert} from "../../../utils/Alert";
+import {Validators, FormGroup, FormBuilder} from "@angular/forms";
```
Comment (ghiobi) [inline src/pages/manage/withdraw/withdraw.component.ts]: Missing spaces

File: src/pages/manage/withdraw/withdraw.component.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      amount: ['', [Validators.required]],
+      wallet: [null, Validators.required],
+      email: ['', [Validators.required]]
+    });
+
+    this.wallets = this.walletSrv.wallets;
+  }
+
+  ionViewDidLoad() {
+
```
Comment (ghiobi) [inline src/pages/manage/withdraw/withdraw.component.ts]: remove function. unused

File: src/pages/manage/withdraw/withdraw.component.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      );
+
+    } else {
+      new Alert(
+        this.alertCtrl,
+        "Withdrawal Failed",
+        displayAmount + " could not have been processed." + res.errors.message,
+        ["Dismiss"]);
+    }
+
```
Comment (ghiobi) [inline src/pages/manage/withdraw/withdraw.component.ts]: weird spacing

File: src/pages/send-money/send-money.html [hunk 7]
```diff
@@ -14,47 +9,51 @@
...
+          <ion-select formControlName="receiver">
+            <ion-option value="johndoe">John Doe</ion-option>
+            <ion-option value="janedoe">Jane Doe</ion-option>
+            <ion-option value="trump">Donald Trump (therealdonaldtrump@gmail.com)</ion-option>
+          </ion-select>
+        </ion-item>
+      </ion-card-content>
+    </ion-card>
+
+
```
Comment (ghiobi) [inline src/pages/send-money/send-money.html]: too many spaces

File: src/pages/send-money/send-money.html [hunk 8]
```diff
@@ -14,47 +9,51 @@
...
+  <ion-card>
+    <ion-card-content>
+      <ion-item>
+        <ion-label stacked>Message</ion-label>
+        <ion-textarea rows="4" formControlName="message" placeholder="Write a message (optional)."  maxlength="255"></ion-textarea>
+      </ion-item>
+    </ion-card-content>
+  </ion-card>
+
+
```
Comment (ghiobi) [inline src/pages/send-money/send-money.html]: too many spaces

File: src/pages/send-money/send-money.ts [hunk 9]
```diff
@@ -1,24 +1,131 @@
-import { Component } from "@angular/core";
-import { NavController, NavParams } from "ionic-angular";
+import {Component, OnInit} from '@angular/core';
+import {NavController, NavParams, AlertController, LoadingController, Loading} from 'ionic-angular';
```
Comment (ghiobi) [inline src/pages/send-money/send-money.ts]: spaces missing

File: src/pages/send-money/send-money.ts [hunk 10]
```diff
@@ -1,24 +1,131 @@
...
+    public navParams: NavParams,
+    public transfSrv: TransferService,
+    public walletSrv: WalletsService,
+    public alertCtrl: AlertController,
+    private formBuilder: FormBuilder,
+    public loadingCtrl: LoadingController
 
-  constructor(public navCtrl: NavController, public navParams: NavParams) {}
+  )
+  {
```
Comment (ghiobi) [inline src/pages/send-money/send-money.ts]: bring bracket up `) {`

File: src/pages/send-money/send-money.ts [hunk 11]
```diff
@@ -1,24 +1,131 @@
...
+      amount: ['', [Validators.required]],
+      wallet: [null, Validators.required],
+      receiver: ['', Validators.required],
+      message: ['', Validators.maxLength(255)]
+    });
+
+    this.wallets = this.walletSrv.wallets;
+  }
 
   ionViewDidLoad() {
```
Comment (ghiobi) [inline src/pages/send-money/send-money.ts]: Remove unused function

File: src/providers/transfer.service.ts [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+          observer.next(res);
+          observer.complete();
+        });
+    });
+    return data;
+  }
+
+
+  public receive() {
```
Comment (ghiobi) [inline src/providers/transfer.service.ts]: missing TsDoc

File: src/utils/Alert.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import {App,AlertController} from "ionic-angular";
+import {Inject} from "@angular/core";
```
Comment (ghiobi) [inline src/utils/Alert.ts]: spaces.

File: src/utils/Alert.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+export class Alert {
+  constructor(
+    private alertCtrl: AlertController,
+    private title: string,
+    private message: string,
+    private buttons: any[]
+  )
+  {
```
Comment (ghiobi) [inline src/utils/Alert.ts]: bring bracket up

File: src/utils/Alert.ts [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    alertCtrl.create({title: title, message: message, buttons: buttons}).present();
+  }
+
```
Comment (ghiobi) [inline src/utils/Alert.ts]: too many spaces.

File: src/providers/transfer.service.ts [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    let data = new Observable(observer => {
+      this.http.post(environment.server_url + "/api/transfer/user/send", {
+        receiver: receiver,
+        amount: amount,
+        wallet_id: wallet_id,
+        message: message
+      })
+        .map(res => res.json())
+        .subscribe(res => {
+
```
Comment (ghiobi) [inline src/providers/transfer.service.ts]: remove extra space

File: src/pages/manage/withdraw/withdraw.component.ts [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+    if (!this.form.value.wallet) {
+      return;
+    }
+    else {
+      this.validAmount = (this.form.value.amount <= parseFloat(this.form.value.wallet.balance));
+      this.maxAmount = this.form.value.wallet.balance;
+      this.maxCurrency = this.form.value.wallet.currency_code;
+    }
+  }
+
```
Comment (ghiobi) [inline src/pages/manage/withdraw/withdraw.component.ts]: remove a space

File: src/pages/manage/manage.ts [hunk 18]
```diff
@@ -15,161 +14,20 @@ export class ManagePage {
...
-                this.alertCtrl.create({
-                  title: "Withdrawal Failed",
-                  subTitle: displayAmount + " could not have been processed.",
-                  buttons: ["Dismiss"],
-                }).present();
-
+  private transferErrors: Error;
+
+  constructor(public navCtrl: NavController,
+              public navParams: NavParams,
```
Comment (ghiobi) [inline src/pages/manage/manage.ts]: weird indentation
```

constructor(
   public navCtrl: NavController,
   public navParams: NavParams,
   public bankSrv: BankTransferService,
   public storage: Storage,
   ...
```

**Review by ghiobi — CHANGES_REQUESTED** [review]
Please practice on dividing up tasks for pull requests.

File: src/pages/manage/manage.html [hunk 19]
```diff
@@ -26,92 +26,12 @@
...
-            <ion-input [(ngModel)]="addMoney.amount"
-                       type="number"
-                       placeholder="{{ addMoney.decimalPlaces }}"
-                       step="0.01"
-                       min="0"
-            ></ion-input>
-          </ion-item>
-          <ion-item>
-            <ion-label stacked>Select Currency</ion-label>
-            <ion-select [(ngModel)]="addMoney.currency" (ionChange)="setDecimalPlaces()">
```
Comment (Scieon) [inline src/pages/manage/manage.html:45]: Missing in the refactored deposit.html

**Review by Scieon — DISMISSED** [review]

**Comment by Scieon** [comment]
In the future lets work on getting smaller pull requests, the first commit was on March 3 and now its become very large. There are conflicts that have to be addressed with current version-2.0. Conflicts will not be displayed here in the case of refactoring so there's a lot of manual checking to be done which is inconvenient.

**Review by ghiobi — DISMISSED** [review]
Thank you james. Good work.

File: src/pages/manage/manage.html (hunk 19 above)
Comment (JTalarico) [reply src/pages/manage/manage.html:45]: It's in the withdraw.component.html.

The manage page was split into two components. Withdraw and deposit

Comment (Scieon) [reply src/pages/manage/manage.html:45]: The stuff that was done in my latest commit has been removed in this branch

Comment (JTalarico) [reply src/pages/manage/manage.html:45]: When I pull version2 it says already up to date

**Review by Scieon — APPROVED** [review]

**Pull request merged**